diff --git a/src/Makevars b/src/Makevars
index 9a39cbe..49e5b67 100644
--- a/src/Makevars
+++ b/src/Makevars
@@ -1,17 +1,21 @@
-# Get the current directory path of the Makevars
-CURRENT_DIR := $(CURDIR)
-PARENT_DIR  := $(shell dirname $(CURRENT_DIR))
-# Get the parent of the parent directory path
-PARENT_OF_PARENT_DIR := $(shell dirname $(PARENT_DIR))
-ephoto_path := $(PARENT_OF_PARENT_DIR)/ePhotosynthesis_C
-
-conda_path := $(shell conda env list | grep ephoto | awk '{print $$NF}')
-
-PKG_CPPFLAGS += -I$(ephoto_path)/include -I$(conda_path)/include
 PKG_CPPFLAGS += -DR_NO_REMAP -O3 -g -Wall -ggdb3
 
-#YH: Both PKG_LIBS and the system environment need the ephoto's library path 
-PKG_LIBS += -L$(ephoto_path)/build -lEPhotosynthesis
+#ML: Newest version of boost requires C++14
+CXX_STD = CXX14
+
+#ML: Assumes ePhotosynthesis is installed in the current conda env
+
+conda_build := ${CONDA_BUILD}
+ifdef conda_build
+conda_path := ${PREFIX}
+else
+conda_path := ${CONDA_PREFIX}
+endif
+ifdef conda_path
+PKG_CPPFLAGS += -I$(conda_path)/include -I$(conda_path)/include/ePhotosynthesis
+PKG_LIBS += -L$(conda_path)/lib
+endif
+PKG_LIBS += -lEPhotosynthesis
 
 SOURCES = $(wildcard *.cpp module_library/*.cpp framework/*.cpp framework/ode_solver_library/*.cpp framework/utils/*.cpp)
 OBJECTS = $(SOURCES:.cpp=.o)
diff --git a/src/Makevars.win b/src/Makevars.win
index a7747e9..454296f 100644
--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -8,7 +8,20 @@
 # If the version of make included with Rtools is updated to version 3.80 or later,
 # then the file will likely be unnecessary.
 
-PKG_CPPFLAGS+=-I../inc -DR_NO_REMAP
+PKG_CPPFLAGS+=-I../inc -DR_NO_REMAP -O3 -g -Wall -ggdb3
+
+#ML: Assumes ePhotosynthesis is installed in the current conda env
+conda_build := ${CONDA_BUILD}
+ifdef conda_build
+conda_path := ${PREFIX}
+else
+conda_path := ${CONDA_PREFIX}
+endif
+ifdef conda_path
+PKG_CPPFLAGS += -I$(conda_path)/include -I$(conda_path)/include/ePhotosynthesis
+PKG_LIBS += -L$(conda_path)/lib
+endif
+PKG_LIBS += -lEPhotosynthesis
 
 SOURCES = $(wildcard *.cpp module_library/*.cpp framework/*.cpp framework/ode_solver_library/*.cpp framework/utils/*.cpp)
 OBJECTS = $(SOURCES:.cpp=.o)
diff --git a/src/module_library/c3photo.cpp b/src/module_library/c3photo.cpp
index b2b6214..baff01a 100644
--- a/src/module_library/c3photo.cpp
+++ b/src/module_library/c3photo.cpp
@@ -12,21 +12,9 @@
 #include <sstream>
 #include <vector>
 #include <iterator>
-#include <boost/algorithm/string_regex.hpp>
-#include <boost/regex.hpp>
-#include "cxxopts.hpp"
-#include "globals.hpp"
-#include "modules/trDynaPS.hpp"
-#include "modules/CM.hpp"
-#include "modules/EPS.hpp"
-#include "modules/PR.hpp"
-#include "drivers/drivers.hpp"
-#include "Variables.hpp"
-
-const boost::regex token("\\s+");
+#include <ePhotosynthesis/ePhotosynthesis.hpp>
+
 using namespace ePhotosynthesis;
-using namespace ePhotosynthesis::drivers;
-using namespace ePhotosynthesis::modules;
 
 //This is now using the ephoto function
 //the original FvCB has been renamed to c3photoC_FvCB
@@ -144,8 +132,8 @@ struct c3_str c3photoC(
 ephoto_Result assim_ephoto(double LeafT, double PAR, double Ci,double enzyme_sf)
 {
 //enzyme_sf is no longer used. Kept as dummy variable for now
-        bool record    = true;//turn this on to calculate penalties
-        bool runBioCro = true;//turn this on to NOT substract the penalties from A
+        // bool record    = true;//turn this on to calculate penalties
+        // bool runBioCro = true;//turn this on to NOT substract the penalties from A
         double stoptime=5000.0, begintime=0.0, stepsize=0.5;
         int  maxSubSteps=3000;
 //        double abstol = 9.9e-6, reltol = 1e-4;
@@ -155,91 +143,29 @@ ephoto_Result assim_ephoto(double LeafT, double PAR, double Ci,double enzyme_sf)
         std::string atpcost="InputATPCost.txt";
         std::string enzymeFile="Einput7.txt";
         std::map<std::string, std::string> inputs;
-
-        readFile1(evn_input, inputs);
-        readFile1(atpcost, inputs);
-
-        Variables *theVars = new Variables();
-
-        readFile2(enzymeFile, theVars->EnzymeAct);
-
-        theVars->CO2_in = Ci;  //get value from BioCro 
-        theVars->TestLi = PAR; //get value from BioCro 
-        if (stoi(inputs.at("SucPath"), nullptr) > 0)
-            CM::setTestSucPath(true);
-        theVars->TestATPCost = stoi(inputs.at("ATPCost"), nullptr);
-        theVars->record = record;
-        theVars->runBioCro = runBioCro;
-        theVars->useC3 = true;
-        theVars->RUBISCOMETHOD = 2;
-	PR::setRUBISCOTOTAL(3);
-//this is the scaling factor for some enzymes. check EPS_Drive.cpp to see
-//which enzymes are being scaled
-//        theVars->sensitivity_sf = enzyme_sf;
-
-        Driver *maindriver;
-       
-//        std::cout<< "leafT is "<<LeafT<<std::endl;
-//        std::cout<< "PAR is "<<theVars->TestLi<<std::endl;
-//        std::cout<< "co2 is "<<theVars->CO2_in<<std::endl;
-//        std::cout<< "alpha2 is"<<theVars->alpha2<<std::endl;
-//        std::cout<< "JMAX is"<<theVars->EnzymeAct.at("Jmax")<<std::endl;
-
-        maindriver = new EPSDriver(theVars, begintime, stepsize, stoptime, maxSubSteps, abstol, reltol, 1, 1, LeafT);
-
-        std::vector<double> ResultRate = maindriver->run();
-
-       if (theVars != nullptr) {
-           maindriver->inputVars= nullptr;
-           delete theVars;
-       }   
-       delete maindriver;
-       
+        inputs["CO2"] = Ci;
+        inputs["PAR"] = PAR;
+        inputs["RUBISCOMETHOD"] = 2;
+        inputs["RUBISCOTOTAL"] = 3;
+        inputs["Tp"] = LeafT;
+        // inputs["record"] = record;
+        // inputs["runBioCro"] = runBioCro;
+        // // This is the scaling factor for some enzymes
+        // // inputs["sensitivity_sf"] = enzyme_sf;
+        
+        std::map<std::string, double> results_map;
+        results_map = run_simulation(
+            (DriverType)4, begintime, stoptime, stepsize, maxSubSteps,
+            abstol, reltol, inputs, true,
+            evn_input, atpcost, enzymeFile
+            // , "", "", {}, 0, "", {}, 0, false, false, record,
+                                     );
        ephoto_Result result; 
-       result.A       = ResultRate[0];
-       result.penalty = ResultRate[3];
+       result.A       = results_map["CO2AR"]; // ResultRate[0];
+       // result.penalty = results_map["penalty"]; // ResultRate[3];
        return result;
 }
 
-void readFile1(const std::string &filename, std::map<std::string, std::string> &mapper) {
-    std::vector<std::string> tempVec;
-    std::string input;
-    std::ifstream inputfile(filename);
-    if(inputfile.fail()) {
-        std::cout << "Could not open " << filename << " for reading" << std::endl;
-        exit(EXIT_FAILURE);
-    }
-    while (getline(inputfile, input)) {
-        if (input.empty())
-            return;
-        boost::algorithm::split_regex(tempVec, input, token);
-        mapper.insert(std::pair<std::string, std::string>(tempVec[0], tempVec[1]));
-    }
-}
-
-void readFile2(const std::string &filename, std::map<std::string, double> &mapper) {
-    std::vector<std::string> tempVec;
-    std::string input;
-    std::ifstream inputfile(filename);
-    if(inputfile.fail()) {
-        std::cout << "Could not open " << filename << " for reading" << std::endl;
-        exit(EXIT_FAILURE);
-    }
-    int count = 0;
-    while (getline(inputfile, input)) {
-        if (input.empty())
-            return;
-        boost::algorithm::split_regex(tempVec, input, token);
-        double d;
-        std::stringstream ss(tempVec[1]);
-        ss >> d;
-        if (count < 27)
-            d /= 30.;
-        count++;
-        mapper.insert(std::pair<std::string, double>(tempVec[0], d));
-    }
-}
-
 struct c3_str c3photoC_FvCB(
     double const Qp,                           // micromol / m^2 / s
     double const Tleaf,                        // degrees C
diff --git a/src/module_library/c3photo.hpp b/src/module_library/c3photo.hpp
index 7f4fcfa..da89865 100644
--- a/src/module_library/c3photo.hpp
+++ b/src/module_library/c3photo.hpp
@@ -62,7 +62,4 @@ struct ephoto_Result {
 };
 ephoto_Result assim_ephoto(double LeafT,double PAR, double Ci,double enzyme_sf);
 
-void readFile1(const std::string &filename, std::map<std::string,std::string> &mapper);
-void readFile2(const std::string &filename, std::map<std::string, double> &mapper); 
-
 #endif
