diff --git a/src/Makevars b/src/Makevars
index 9a39cbe..02183fc 100644
--- a/src/Makevars
+++ b/src/Makevars
@@ -1,17 +1,12 @@
-# Get the current directory path of the Makevars
-CURRENT_DIR := $(CURDIR)
-PARENT_DIR  := $(shell dirname $(CURRENT_DIR))
-# Get the parent of the parent directory path
-PARENT_OF_PARENT_DIR := $(shell dirname $(PARENT_DIR))
-ephoto_path := $(PARENT_OF_PARENT_DIR)/ePhotosynthesis_C
-
-conda_path := $(shell conda env list | grep ephoto | awk '{print $$NF}')
-
-PKG_CPPFLAGS += -I$(ephoto_path)/include -I$(conda_path)/include
 PKG_CPPFLAGS += -DR_NO_REMAP -O3 -g -Wall -ggdb3
 
-#YH: Both PKG_LIBS and the system environment need the ephoto's library path 
-PKG_LIBS += -L$(ephoto_path)/build -lEPhotosynthesis
+#ML: Assumes ePhotosynthesis is installed in the current conda env
+conda_path := ${CONDA_PREFIX}
+ifdef conda_path
+PKG_CPPFLAGS += -I$(conda_path)/include -I$(conda_path)/include/ePhotosynthesis
+PKG_LIBS += -L$(conda_path)/lib
+endif
+PKG_LIBS += -lEPhotosynthesis
 
 SOURCES = $(wildcard *.cpp module_library/*.cpp framework/*.cpp framework/ode_solver_library/*.cpp framework/utils/*.cpp)
 OBJECTS = $(SOURCES:.cpp=.o)
diff --git a/src/Makevars.win b/src/Makevars.win
index a7747e9..63b5733 100644
--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -8,7 +8,15 @@
 # If the version of make included with Rtools is updated to version 3.80 or later,
 # then the file will likely be unnecessary.
 
-PKG_CPPFLAGS+=-I../inc -DR_NO_REMAP
+PKG_CPPFLAGS+=-I../inc -DR_NO_REMAP -O3 -g -Wall -ggdb3
+
+#ML: Assumes ePhotosynthesis is installed in the current conda env
+conda_path := ${CONDA_PREFIX}
+ifdef conda_path
+PKG_CPPFLAGS += -I$(conda_path)/include -I$(conda_path)/include/ePhotosynthesis
+PKG_LIBS += -L$(conda_path)/lib
+endif
+PKG_LIBS += -lEPhotosynthesis
 
 SOURCES = $(wildcard *.cpp module_library/*.cpp framework/*.cpp framework/ode_solver_library/*.cpp framework/utils/*.cpp)
 OBJECTS = $(SOURCES:.cpp=.o)
diff --git a/src/module_library/c3photo.cpp b/src/module_library/c3photo.cpp
index b2b6214..6b20f37 100644
--- a/src/module_library/c3photo.cpp
+++ b/src/module_library/c3photo.cpp
@@ -23,7 +23,7 @@
 #include "drivers/drivers.hpp"
 #include "Variables.hpp"
 
-const boost::regex token("\\s+");
+// const boost::regex token("\\s+");
 using namespace ePhotosynthesis;
 using namespace ePhotosynthesis::drivers;
 using namespace ePhotosynthesis::modules;
@@ -144,8 +144,8 @@ struct c3_str c3photoC(
 ephoto_Result assim_ephoto(double LeafT, double PAR, double Ci,double enzyme_sf)
 {
 //enzyme_sf is no longer used. Kept as dummy variable for now
-        bool record    = true;//turn this on to calculate penalties
-        bool runBioCro = true;//turn this on to NOT substract the penalties from A
+        // bool record    = true;//turn this on to calculate penalties
+        // bool runBioCro = true;//turn this on to NOT substract the penalties from A
         double stoptime=5000.0, begintime=0.0, stepsize=0.5;
         int  maxSubSteps=3000;
 //        double abstol = 9.9e-6, reltol = 1e-4;
@@ -168,8 +168,8 @@ ephoto_Result assim_ephoto(double LeafT, double PAR, double Ci,double enzyme_sf)
         if (stoi(inputs.at("SucPath"), nullptr) > 0)
             CM::setTestSucPath(true);
         theVars->TestATPCost = stoi(inputs.at("ATPCost"), nullptr);
-        theVars->record = record;
-        theVars->runBioCro = runBioCro;
+        // theVars->record = record;
+        // theVars->runBioCro = runBioCro;
         theVars->useC3 = true;
         theVars->RUBISCOMETHOD = 2;
 	PR::setRUBISCOTOTAL(3);
