diff --git a/CMakeLists.txt b/CMakeLists.txt
index 705d302..6788646 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -200,7 +200,7 @@ if(WIN32)
 endif()
 
 target_link_libraries(
-    EPhotosynthesis PUBLIC ${Boost_LIBRARIES} ${SUNDIALS_LIBRARIES}
+    EPhotosynthesis ${Boost_LIBRARIES} ${SUNDIALS_LIBRARIES}
 )
 target_include_directories(
     EPhotosynthesis PUBLIC ${Boost_INCLUDE_DIRS} ${SUNDIALS_INCLUDE_DIRS}
